src/main/java/org/opentripplanner/middleware/tripmonitor/TrustedCompanion.java

```diff
@@ -0,0 +1,148 @@
+            Persistence.otpUsers.replace(relatedUser.id, relatedUser);
+            // Update list of related users.
+            Persistence.otpUsers.replace(dependentUser.id, dependentUser);
+        } else {
+            logMessageAndHalt(request, HttpStatus.BAD_REQUEST_400, "Dependent did not request user to be related!");
+            return null;
+        }
+
+        // TODO: Not sure what is required in the response. For now, returning the updated related user.
+        return relatedUser;
+    }
+
+    public static void manageAcceptDependentEmail(OtpUser dependentUser) {
+        manageAcceptDependentEmail(dependentUser, false);
+    }
+
+    /**
+     * When creating or updating an OTP user, extract a list of newly defined dependents and send an 'accept dependent'
+     * email to each. Then update which dependents have been sent an email so subsequent updates do not trigger
+     * additional emails.
+     */
+    public static void manageAcceptDependentEmail(OtpUser dependentUser, boolean isTest) {
+        if (dependentUser.relatedUsers.isEmpty()) {
+            // No related users defined by dependent.
+            return;
+        }
+
+        dependentUser.relatedUsers
+            .stream()
+            .filter(relatedUser -> !relatedUser.acceptDependentEmailSent)
+            .forEach(relatedUser -> {
+                OtpUser userToReceiveEmail = Persistence.otpUsers.getById(relatedUser.userId);
+                if (userToReceiveEmail != null && (isTest || sendAcceptDependentEmail(dependentUser, userToReceiveEmail))) {
+                    relatedUser.acceptDependentEmailSent = true;
+                }
+            });
+
+        // Preserve email sent status.
+        Persistence.otpUsers.replace(dependentUser.id, dependentUser);
+    }
+
+    /**
+     * Send 'accept dependent' email.
+     */
+    private static boolean sendAcceptDependentEmail(OtpUser dependentUser, OtpUser relatedUser) {
+        Locale locale = I18nUtils.getOtpUserLocale(relatedUser);
+
+        String acceptDependentLinkLabel = Message.ACCEPT_DEPENDENT_EMAIL_LINK_TEXT.get(locale);
+        String acceptDependentUrl = getAcceptDependentUrl(dependentUser);
+
+        // A HashMap is needed instead of a Map for template data to be serialized to the template renderer.
+        Map<String, Object> templateData = new HashMap<>(
+            Map.of(
+                "acceptDependentLinkAnchorLabel", acceptDependentLinkLabel,
+                "acceptDependentLinkLabelAndUrl", label(acceptDependentLinkLabel, acceptDependentUrl, locale),
+                "acceptDependentUrl", getAcceptDependentUrl(dependentUser),
+                "emailFooter", Message.ACCEPT_DEPENDENT_EMAIL_FOOTER.get(locale),
+                // TODO: The user's email address isn't very personal, but that is all I have to work with! Suggetions?
+                "emailGreeting", String.format("%s%s", dependentUser.email, Message.ACCEPT_DEPENDENT_EMAIL_GREETING.get(locale)),
```

Comment:
I think user's email address is fine. More personalized greetings might not localize well. I can also imagine cases where the dependent user (or the caregiver user) are institutional email addresses.